lsp-mode.el

```diff
@@ -5431,13 +5430,54 @@ If ACTION is not set it will be selected from `lsp-code-actions-at-point'."
    ((stringp command?) (lsp--execute-command action))
    ((lsp-command? command?) (lsp--execute-command command?))))
 
+(defvar lsp--formatting-indent-alist
```

Comment:
is there a source for that or you did it by yourself?

Reply:
I took it from `dtrt-indent-mode`.